### ISSUE TITLE
can't redefine a variable after define it as function more than once.

### ISSUE THREAD

**Issue by qtxie** [red/red#831]
This code will fail:
%test.red
```
Red []

f: function [][1]
f: function [][1]

f: 100
print f = 100
```
output:
```
> test.exe
*** Error: word has no value!
false
```

but this one works fine.
```
Red []

f: function [][1]
;; f: function [][1]         ;-- comment it.

f: 100
print f = 100
```
output:
```
true
```

**Comment by dockimbel** [comment]
Works fine now.

**Comment by rebolek** [comment]
Still not fixed, see #2207
